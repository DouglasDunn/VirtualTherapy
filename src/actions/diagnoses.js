import uuid from 'uuid';

// ADD_DIAGNOSES
export const addDiagnoses = (diagnoses) => ({
  type: 'ADD_DIAGNOSES',
  diagnoses
});

export const createDiagnoses = (diagnoses) => {
  return (dispatch) => {
    diagnoses = diagnoses
      .split(',')
      .map(diagnose => {
        const id = uuid.v4();
        return {
          id,
          diagnoses: diagnose.trim(),
          medication: "",
          time: ""
        };
      });
    dispatch(addDiagnoses(diagnoses));
  }
};

// ADD_MEDICATION
export const addEditMedication = (id, medicationObject) => ({
  type: 'ADD_EDIT_MEDICATION',
  id,
  medicationObject
});
