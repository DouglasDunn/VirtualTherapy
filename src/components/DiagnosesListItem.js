import React from 'react';
import { Link } from 'react-router-dom';

const DiagnosesListItem = ({ id, diagnoses, medication, time }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{diagnoses}</h3>
      <span className="list-item__sub-title">{medication}</span>
    </div>
    <h3 className="list-item__data">{time}</h3>
  </Link>
);

export default DiagnosesListItem;
