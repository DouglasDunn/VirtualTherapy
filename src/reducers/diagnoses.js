// Diagnoses Reducer

const diagnosesReducerDefaultState = [];

export default (state = diagnosesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DIAGNOSES':
      return [
        ...state,
        action.diagnoses
      ];
    case 'SET_DIAGNOSES':
      return action.diagnoses;
    default:
      return state;
  }
};
