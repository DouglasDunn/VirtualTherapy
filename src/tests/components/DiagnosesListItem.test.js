import React from 'react';
import { shallow } from 'enzyme';
import diagnoses from '../fixtures/diagnoses';
import DiagnosesListItem from '../../components/DiagnosesListItem';

test('should render DiagnosesListItem correctly', () => {
  const wrapper = shallow(<DiagnosesListItem {...diagnoses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
