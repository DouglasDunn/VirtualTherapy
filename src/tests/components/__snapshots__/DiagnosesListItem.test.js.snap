// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DiagnosesListItem correctly 1`] = `
<Link
  className="list-item"
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      clinical depression
    </h3>
    <span
      className="list-item__sub-title"
    >
      zoloft
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    9:30pm
  </h3>
</Link>
`;
